src/Tools/dotnet-monitor/Startup.cs

```diff
@@ -30,15 +28,14 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddMvc((MvcOptions options) =>
-            {
-                options.Filters.Add(new ProducesAttribute("application/json"));
-
-                // HACK We need to disable EndpointRouting in order to run properly in 3.1
-                System.Reflection.PropertyInfo prop = options.GetType().GetProperty("EnableEndpointRouting");
-                prop?.SetValue(options, false);
-            }).SetCompatibilityVersion(CompatibilityVersion.Latest);
+            services.AddMvc(options =>
+                {
+                    options.Filters.Add(new ProducesAttribute("application/json"));
 
+                    options.EnableEndpointRouting = false;
```

Comment:
The default settings for 3.1 disallow async apis:
System.InvalidOperationException: 'Synchronous operations are disallowed. Call WriteAsync or set AllowSynchronousIO to true instead.'

Can you test out /logs scenarios with event-stream and json to make sure these still work?

Reply:
Yeah, that still happens. I could allow sync IO in dotnet-monitor, which would temporarily solve this, but is probably not what we want long term when used as a library in a 3rd party application.

Reply:
The long term solution may be a bit complex and look like ConsoleLogger's Log calls, with a queue to handle messages. Alternatively, we need to switch out of ILogger for consumption of log events.

Reply:
As discussed offline, we'll allow synchronous IO for now until a replacement for the StreamingLogger is designed.